modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/service/OnboardServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.modelservice.service;
+
+import com.expedia.adaptivealerting.modelservice.entity.Metric;
+import com.expedia.adaptivealerting.modelservice.entity.MetricTagMapping;
+import com.expedia.adaptivealerting.modelservice.entity.Tag;
+import com.expedia.adaptivealerting.modelservice.repo.ItemExistsException;
+import com.expedia.adaptivealerting.modelservice.repo.MetricRepository;
+import com.expedia.adaptivealerting.modelservice.repo.MetricTagMappingRepository;
+import com.expedia.adaptivealerting.modelservice.repo.TagRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * OnboardServiceDetail Implementation
+ *
+ * @author tbahl
+ */
+
+
+@Service
+@Slf4j
+public class OnboardServiceImpl implements OnboardService {
+
+    @Autowired
+    private TagRepository tagRepository;
+
+
+    @Autowired
+    private MetricRepository metricRepository;
+
+    @Autowired
+    private MetricTagMappingRepository metricTagMappingRepository;
+
+
+    @Override
+    public Long onboarded(Metric metric) {
+        String hash = metric.getHash();
+        Metric m = metricRepository.findByHash(hash);
+        System.out.println("Metric is: "+m);
+        if (m != null) {
+            System.out.println("Metric Exist");
+            throw new ItemExistsException(m);
+        }
+        List<Tag> tagList;
+        Metric metricEntry;
+        Tag tagEntry;
+        Long metricID = Long.valueOf(0);
```

Comment:
Move all the variables to the top of the function and initialize list and objects here.